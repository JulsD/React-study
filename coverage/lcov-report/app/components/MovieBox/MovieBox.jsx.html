<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/MovieBox/MovieBox.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">app/components/MovieBox</a> MovieBox.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">90% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { connect } from 'react-redux';
&nbsp;
import styles from './movie-box.css';
&nbsp;
export class MovieBox extends React.Component {
&nbsp;
  constructor(props) {
    super(props);
    this.state = {} 
  }
&nbsp;
  render() {
    let movie = this.props.movie;
    const imageBaseUrl = 'https://image.tmdb.org/t/p/w500';
    let coverPath = movie.poster_path ? imageBaseUrl + movie.poster_path : <span class="branch-1 cbranch-no" title="branch not covered" >''</span>
    return (
      &lt;div className = {styles.root}&gt;
        &lt;div className = {styles.coverWrapper}&gt;
          &lt;div className = {styles.cover}&gt;
            &lt;div style={{backgroundImage: 'url(' + coverPath + ')'}}&gt;&lt;/div&gt;
          &lt;/div&gt;
        &lt;/div&gt;
        &lt;div className = {styles.details}&gt;
          &lt;div className = {styles.row}&gt;
            &lt;h2 className = {styles.title}&gt;
              {movie.title}
            &lt;/h2&gt;
            &lt;div className = {styles.rate}&gt;
              &lt;span&gt;{movie.vote_average}&lt;/span&gt;
            &lt;/div&gt;
            &lt;p className = {styles.genre}&gt;
              {movie.genres.map(item =&gt; &lt;span key={item.id}&gt;{item.name} &lt;/span&gt;)}
            &lt;/p&gt;
            &lt;div className = {styles.numb}&gt;{movie.release_date}&lt;/div&gt;
            &lt;div className = {styles.numb}&gt;{movie.runtime} min&lt;/div&gt;
          &lt;/div&gt;
          &lt;div className = {styles.row}&gt;
            &lt;p className = {styles.description}&gt;{movie.overview}&lt;/p&gt;
          &lt;/div&gt;
          &lt;div className = {styles.additional}&gt;
            &lt;p&gt;
              {movie.director.map(item =&gt; &lt;span key={item.id}&gt;{item.name}&lt;i&gt;, &lt;/i&gt;&lt;/span&gt;)}
            &lt;/p&gt;
            &lt;p&gt;
              {movie.actors.map(item =&gt; &lt;span key={item.cast_id}&gt;{item.name}&lt;i&gt;, &lt;/i&gt;&lt;/span&gt;)}
                
            &lt;/p&gt;
          &lt;/div&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }
&nbsp;
}
&nbsp;
function <span class="fstat-no" title="function not covered" >mapProps(</span>state) {
<span class="cstat-no" title="statement not covered" >  return {</span>
    movie: state.movie.movie
  }
}
&nbsp;
export default connect(mapProps)(MovieBox);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 09 2017 00:12:39 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
