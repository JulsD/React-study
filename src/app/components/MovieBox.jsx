import React from 'react';

class MovieBox extends React.Component {

  constructor(props) {
    super(props);

  }

  render() {
    const movieTitle = this.props.title;
    return (
      <div className="movie-card">
        <div className="movie-cover"></div>
        <div className="movie-details">
          <div className="movie-title">{this.props.title} Title</div>
          <div className="movie-rate">1990</div>
          <div className="movie-genre">genre</div>
          <div className="movie-date">1990</div>
          <div className="movie-duratiom">{154} min</div>
          <div className="movie-description">description</div>
          <div className="movie-director">director</div>
          <div className="movie-cast-list">cast-list</div>
        </div>
      </div>
    );
  }

}

export default MovieBox;
