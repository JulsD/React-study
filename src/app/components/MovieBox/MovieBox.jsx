import React from 'react';
import { connect } from 'react-redux';

import styles from './movie-box.css';

export class MovieBox extends React.Component {

  constructor(props) {
    super(props);
    this.state = {} 
  }

  render() {
    let movie = this.props.movie;
    const imageBaseUrl = 'https://image.tmdb.org/t/p/w500';
    let coverPath = movie.poster_path ? imageBaseUrl + movie.poster_path : ''
    return (
      <div className = {styles.root}>
        <div className = {styles.coverWrapper}>
          <div className = {styles.cover}>
            <div style={{backgroundImage: 'url(' + coverPath + ')'}}></div>
          </div>
        </div>
        <div className = {styles.details}>
          <div className = {styles.row}>
            <h2 className = {styles.title}>
              {movie.title}
            </h2>
            <div className = {styles.rate}>
              <span>{movie.vote_average}</span>
            </div>
            <p className = {styles.genre}>
              {movie.genres.map(item => <span key={item.id}>{item.name} </span>)}
            </p>
            <div className = {styles.numb}>{movie.release_date}</div>
            <div className = {styles.numb}>{movie.runtime} min</div>
          </div>
          <div className = {styles.row}>
            <p className = {styles.description}>{movie.overview}</p>
          </div>
          <div className = {styles.additional}>
            <p>
              {movie.director.map(item => <span key={item.id}>{item.name}<i>, </i></span>)}
            </p>
            <p>
              {movie.actors.map(item => <span key={item.cast_id}>{item.name}<i>, </i></span>)}
                
            </p>
          </div>
        </div>
      </div>
    );
  }

}

function mapProps(state) {
  return {
    movie: state.movie.movie
  }
}

export default connect(mapProps)(MovieBox);
