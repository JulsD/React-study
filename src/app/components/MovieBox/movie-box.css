.root {
  width: 100%;
  display: flex;
  padding: 30px 0;
}
.coverWrapper {
  flex: 0 0 33%;
}
.cover {
  position: relative;
  width: 100%;
  padding-top: 130%;
}
.cover div {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  background-color: rgba(255,255,255,.5);
}
.details {
  padding: 10px 5px;
  margin-left: 40px;
  color: rgba(255, 255, 255, .9);
  font-size: 14px;
  flex: 1 1 auto;
}
.row {
  margin-bottom: 30px;
}
.title {
  text-transform: capitalize;
  font-size: 30px;
  font-weight: 700;
  color: #f40612;
  display: inline-block;
  vertical-align: middle;
  margin: 0;
}
.rate {
  width: 40px;
  height: 40px;
  padding: 5px;
  box-sizing: border-box;
  text-align: center;
  border: 1px solid #00b200;
  color: #00b200;
  border-radius: 50%;
  display: inline-block;
  vertical-align: middle;
  margin: 0 20px;
  font-size: 18px;
  position: relative;
}
.rate span {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
.genre {
  margin-bottom: 15px;
}
.numb {
  font-weight: 600;
  display: inline-block;
  vertical-align: middle;
  margin-right: 20px;
  color: #fff;
}
.additional {
  font-size: 12px;
}
.additional p {
  margin-bottom: 10px;
}
