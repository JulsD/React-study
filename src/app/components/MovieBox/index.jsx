import { default as MovieBox } from './MovieBox';
export default MovieBox;