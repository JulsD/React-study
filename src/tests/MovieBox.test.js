import React from 'react';
import { MovieBox } from '../app/components/MovieBox/MovieBox';

test('MovieBox should render a <div>', () => {
  const movie = {
    poster_path: 'test',
    title: 'test',
    vote_average: 1,
    genres: [
      { id: 1, name: 'test' }, 
      { id: 2, name: 'test'}
    ],
    release_date: 'test',
    runtime:'test',
    overview: 'test',
    director: [
      { id: 1, name: 'test' }, 
      { id: 2, name: 'test'}
    ],
    actors: [
      { cast_id: 1, name: 'test' }, 
      { cast_id: 2, name: 'test'}
    ]
  }
  const wrapper = shallow(
    <MovieBox movie={movie}/>
  );
  expect(wrapper).toMatchSnapshot();
});
