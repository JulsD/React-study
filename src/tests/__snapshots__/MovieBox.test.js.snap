// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieBox should render a <div> 1`] = `
<div
  className="root"
>
  <div
    className="coverWrapper"
  >
    <div
      className="cover"
    >
      <div
        style={
          Object {
            "backgroundImage": "url(https://image.tmdb.org/t/p/w500test)",
          }
        }
      />
    </div>
  </div>
  <div
    className="details"
  >
    <div
      className="row"
    >
      <h2
        className="title"
      >
        test
      </h2>
      <div
        className="rate"
      >
        <span>
          1
        </span>
      </div>
      <p
        className="genre"
      >
        <span
          key="1"
        >
          test
           
        </span>
        <span
          key="2"
        >
          test
           
        </span>
      </p>
      <div
        className="numb"
      >
        test
      </div>
      <div
        className="numb"
      >
        test
         min
      </div>
    </div>
    <div
      className="row"
    >
      <p
        className="description"
      >
        test
      </p>
    </div>
    <div
      className="additional"
    >
      <p>
        <span
          key="1"
        >
          test
          <i>
            , 
          </i>
        </span>
        <span
          key="2"
        >
          test
          <i>
            , 
          </i>
        </span>
      </p>
      <p>
        <span
          key="1"
        >
          test
          <i>
            , 
          </i>
        </span>
        <span
          key="2"
        >
          test
          <i>
            , 
          </i>
        </span>
      </p>
    </div>
  </div>
</div>
`;
